fsspec/implementations/dropboxdrivefs.py

```diff
@@ -0,0 +1,239 @@
+        block_size="default",
+        autocommit=True,
+        cache_type="readahead",
+        cache_options=None,
+        **kwargs
+    ):
+        """
+        Open a file.
+        Parameters
+        ----------
+        fs: instance of DropboxDriveFileSystem
+        path : str
+            file path to inspect in dropbox
+        mode: str
+            Normal file modes.'rb' or 'wb'
+        block_size: int or None
+            The amount of read-ahead to do, in bytes. Default is 5MB, or the value
+            configured for the FileSystem creating this file
+        """
+        for key, value in kwargs.items():
+            print("{0} = {1}".format(key, value))
+        self.session = fs.session if fs.session is not None else requests.Session()
+        super().__init__(fs=fs, path=path, mode=mode, block_size=block_size, **kwargs)
+
+        self.path = path
+        self.dbx = self.fs.dbx
+        while "//" in path:
+            path = path.replace("//", "/")
+        self.url = self.dbx.files_get_temporary_link(path).link
+
+    def read(self, length=-1):
+        """Read bytes from file
+
+        Parameters
+        ----------
+        length: int
+            Read up to this many bytes. If negative, read all content to end of
+            file. If the server has not supplied the filesize, attempting to
+            read only part of the data will raise a ValueError.
+        """
+        if (
+            (length < 0 and self.loc == 0)
+            or (length > (self.size or length))  # explicit read all
+            or (  # read more than there is
+                self.size and self.size < self.blocksize
+            )  # all fits in one block anyway
+        ):
+            self._fetch_all()
+        if self.size is None:
+            if length < 0:
+                self._fetch_all()
+        else:
+            length = min(self.size - self.loc, length)
+        return super().read(length)
+
+    def _fetch_all(self):
+        """Read whole file in one shot, without caching
+
+        This is only called when position is still at zero,
+        and read() is called without a byte-count.
+        """
+        if not isinstance(self.cache, AllBytes):
+            r = self.session.get(self.url, **self.kwargs)
+            r.raise_for_status()
+            out = r.content
+            self.cache = AllBytes(out)
+            self.size = len(out)
+
+    def _fetch_range(self, start, end):
+        """Download a block of data
+
+        The expectation is that the server returns only the requested bytes,
+        with HTTP code 206. If this is not the case, we first check the headers,
+        and then stream the output - if the data size is bigger than we
+        requested, an exception is raised.
+        """
+        kwargs = self.kwargs.copy()
+        headers = kwargs.pop("headers", {})
+        headers["Range"] = "bytes=%i-%i" % (start, end - 1)
+        r = self.session.get(self.url, headers=headers, stream=True, **kwargs)
```

Comment:
Much of this seems to be copied from HTTPFile - which is OK, I guess, but less duplication would be good

Reply:
In a previous comment, I asked you how to proceed with this "copy-past". I did not know if I should import http implementation to use this method and avoid redundancy or if copy-past was fine as the implementations should stay independent between them. 
Do you see another way to go ? and if not, which solution do you prefer? Thanks

Reply:
I believe you could have subclassed HTTPFile successfully to reduce the amount of code here, but without trying it, I cannot be sure - you may have had to override many of the methods anyway. Initially I suspected that you could just *use* HTTPFile directly, passing the known size in the constructor.

Reply:
When I try to subclassed HTTPFile, I had two problems:
- the mode error raise: can be fixed with a workaround making the code very ugly
- the info method called with the temporary url path instead of the dropbox path => crashed the code and I couldn't find another workaround. 
Even if I find a workaround, we definitly cannot call this an improvement. The code is unreadable and very much complicated for nothing. 

When I try to directly used an HTTPFile, I add again the info method problem that I don't see how to fix.

Reply:
Indeed, both methods are working. Except, that by subclassed HTTPFile, a little trick is needed to avoid error raise by the mode "wb" 
=> send "rb" in super and set again the mode "wb" after (as well as 4 bool variables) 
Like it makes the code little more dirty, I prefer the second solution (using the object HTTPFile object)

Reply:
OK, well we don't like dirty code, so no need to change it.